src/fsql/column_parser.py

```diff
@@ -146,5 +179,18 @@ def generate(self) -> Optional[list[str]]:
         else:
             return None
 
+    @classmethod
+    def from_str(cls, path_description: str, fname: str | None = None):
+        if fname is None:
+            warnings.warn(
+                "Using `fname` directly in the path was deprecated and will be removed in the future. "
+                "To avoid future incompatibilities, replace `FixedColumnsParser.from_str('c1/c2/fname')` "
+                "with `FixedColumnsParser.from_str('c1/c2', fname='fname')`",
+                FutureWarning,
+                stacklevel=3,
+            )
+            path_description, fname = path_description.rsplit("/", 1)
+        return super().from_str(path_description, fname)
```

Comment:
I would like to keep backward compatibility and have a consitent interface. But I am not sure about the future. Do we want to deprecated the original behaviour (it was required for `FixedColumnsParser` to contain `fname` as well)? I can imagine both would stay but it's better to have almost always only a single way how to do it.

Reply:
I do agree with single ways being preferred -- per my comment about grammars, I don't think we even need to single out this as deprecated here, since ideally this _whole_ mechanism would get deprecated...

Reply:
I change this from FutureWarning to just log a warning, which is kind of similar but not the same (I think it's good to try to force users to prefer one single way over the other). If we decide to implement the grammer in the future, I would probably prefer to deprecate whole `from_str` method (with a warning that users should pin their versions to <=1.0), rather then introducing V2 versions, but that something we can discuss later.

Any suggestion for `fname`? Do you think it's ok to keep it? I would rather use something different, but didn't come up with a good name.

Reply:
future warning is good

deprecating from_str and introducing from_grammar or smth like that is totally fine with me, thats kinda what I meant by v2 (I meant v2 _inside_ some classes and functions, not on the module level... that is a pain)

(for `fname` I responded in another thread)